### PR TITLE
LIMIT edge cases

### LINKED ISSUE #5711: LIMIT clause works improperly for edge cases
### Describe the bug

LIMIT clause works improperly for negative value and null

### To Reproduce

- Neg value
```
❯ select * from (select 1 a union all select 2) b order by a limit -1;
+---+
| a |
+---+
| 1 |
| 2 |
+---+
2 rows in set. Query took 0.067 seconds.
```
The query should fail

- Null
```
❯ select * from (select 1 a union all select 2) b order by a limit null;
Error during planning: Unexpected expression for LIMIT clause
```
This should work as if no limits

### Expected behavior

Should work in sync with PG

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by comphead**
# Which issue does this PR close?

Closes #5711.

# Rationale for this change

See #5711

# What changes are included in this PR?
Fixes for LIMIT edge cases

# Are these changes tested?
Yes

# Are there any user-facing changes?
No

### CODE REVIEW

**Review by kazuyukitanimura — APPROVED** [review]
LGTM (non-binding)

File: datafusion/core/tests/sqllogictests/test_files/select.slt [hunk 1]
```diff
@@ -105,4 +105,24 @@ select
 ----
 false true false true true false false true false true true false true true false false true
 
+# select limit clause
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/select.slt:108]: I double checked the results match postgres:

```sql
postgres=# select * from (select 1 a union all select 2) b order by a limit 1;
 a
---
 1
(1 row)

postgres=# select * from (select 1 a union all select 2) b order by a limit -1;
ERROR:  LIMIT must not be negative
postgres=# select * from (select 1 a union all select 2) b order by a limit null;
 a
---
 1
 2
(2 rows)

postgres=# select * from (select 1 a union all select 2) b order by a limit 0;
 a
---
(0 rows)

```

👍

**Review by alamb — APPROVED** [review]
Thank you @comphead

**Review by viirya — APPROVED** [review]
